### PR TITLE
[develop2][poc] decouple test_package from create

### PR DESCRIPTION
**Pull request by memsharded**
Lets see what breaks

### CODE REVIEW

File: conans/test/functional/toolchains/cmake/cmakedeps/test_apple_frameworks.py [hunk 1]
```diff
@@ -251,7 +251,11 @@ def test(self):
                  "test_package/conanfile.py": test_conanfile,
                  'test_package/CMakeLists.txt': test_cmake,
                  "test_package/timer.cpp": timer_cpp})
-    client.run("create . %s" % settings)
+    # TODO: this is a bit weird, it will build the "host" library in the first pass, that will
+    #  be considered the main tested thing, then when executing the "test_package" it will build
+    #  the library, now for "build" context
+    #  other alternatives would be to have 2 separate "test_package"
+    client.run("create . %s --test-package-build=missing" % settings)
```
Comment (lasote) [inline conans/test/functional/toolchains/cmake/cmakedeps/test_apple_frameworks.py]: The `test-package-build` is pretty complex.

File: conan/cli/commands/create.py [hunk 2]
```diff
@@ -46,10 +47,13 @@ def create(conan_api, parser, *args):
...
     out = ConanOutput()
     out.highlight("Exporting the recipe")
-    ref = conan_api.export.export(path=path,
+    ref, conanfile = conan_api.export.export(path=path,
                                   name=args.name, version=args.version,
                                   user=args.user, channel=args.channel,
                                   lockfile=lockfile)
+
+    is_build_require = args.build_require
+    is_build_require = is_build_require or getattr(conanfile, "is_build_require", False)
```
Comment (lasote) [inline conan/cli/commands/create.py]: is/should this `conanfile. is_build_require` somehow related to the `node. is_build_require`?

**Review by lasote — COMMENTED** [review]
Not looking bad! bit complex the test-package-build and we should think about marking the `build-requires` in the recipe, it could be complex too.

File: conan/cli/commands/create.py (hunk 2 above)
Comment (memsharded) [reply conan/cli/commands/create.py]: ``node.is_build_require`` doesn't exist?

**Review by lasote — APPROVED** [review]
Nice!

**Review by czoido — APPROVED** [review]

**Pull request merged**